scripts/install2

```diff
@@ -2,6 +2,74 @@
 
 set -e
 
+dialogGlobalParams="--backtitle Installation --ascii-lines"
+# Ask if module is to be enabled if not defined
+#    askModule <Variable> <Description> [default answer]
+function askModule {
+
+    # Define standard behaviour (default yes)
+    askPrompt="(Y/n)"
+    nonDefaultMatch="Nn"
+    defaultValue=true
+    nonDefaultValue=false
+    dialogParam=""
+
+    # Define alternative behaviour (default no)
+    if [ "$3" == "n" ]; then
+        askPrompt="(N/y)"
+        nonDefaultMatch="Yy"
+        nonDefaultValue=true
+        defaultValue=false
+        dialogParam=" --defaultno "
+    fi
+
+    # This reads variable repersented by the string
+    eval res=\$$1
+
+    if [ "$(checkModule 'WITH_DIALOG')" ]; then
+
+        # Do not stop exec on non 0 return values
+        set +e
+        dialog $dialogGlobalParams $dialogParam --title "$2" --yesno "Install $2?" 6 55
```

Comment:
Can't do `|| true`?

Reply:
`set -e` stops executing whenever the return value is not 0 (it thinks there is an error) and in MOST cases this is the case. 0 for success 1 for failure. Sometimes other numbers .

By adding `|| true` overwrites the return code with 0 (since that's what true returns)

Dialog sets the return code to the selected item.  In the case of Yes/No  0 for Yes 1 for No.  This means that whenever some one selects No  `set -e` will stop the script.  And if we add  `|| true` it will simply overwrite all responses with Yes  (0)

Only solution i have so far is to `set +e` for the duration of the dialog and set it back to `set -e` right after.  If there is a better solution i would welcome it, or at very least detect if -e is set before applying it again after.